### PR TITLE
Issue #8971: ExecutableStatementCount should allow LAMBDA token

### LINKED ISSUE #8971: ExecutableStatementCount should allow LAMBDA token
from discussion at https://github.com/checkstyle/checkstyle/issues/8966

https://checkstyle.org/config_sizes.html#ExecutableStatementCount

```
$ javac Test.java 
$ cat Test.java 
 public class Test {
  java.util.function.Consumer a = (o) -> {
    o.toString(); // 1
    o.toString(); // 2
    o.toString(); // 3
    o.toString(); // 4
    o.toString(); // 5
    o.toString(); // 6
  };
 }

$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="ExecutableStatementCount">
      <property name="max" value="5"/>
    </module>
  </module>
</module>

$ java -jar checkstyle-8.36.2-all.jar -c config.xml Test.java 
Starting audit...
Audit done.
```

Expected: violation over lambda on line 2

### PR DESCRIPTION
**Pull request by shashwatj07**
Resolves #8971

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/ExecutableStatementCountCheckTest.java [hunk 1]
```diff
@@ -75,6 +75,7 @@ public void testMaxZero() throws Exception {
             "67:5: " + getCheckMessage(MSG_KEY, 2, 0),
             "76:5: " + getCheckMessage(MSG_KEY, 2, 0),
             "79:13: " + getCheckMessage(MSG_KEY, 1, 0),
+            "90:29: " + getCheckMessage(MSG_KEY, 1, 0),
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/sizes/ExecutableStatementCountCheckTest.java:78]: it is not that easy.
Please add more test cases with lambdas like `a -> a.trim()` where executable statement count should be 1.

**Review by strkkk — CHANGES_REQUESTED** [review]
Please also fix CI

**Comment by romani** [comment]
@shashwatj07 , please finish PR.

**Comment by romani** [comment]
@shashwatj07 , ping

**Pull request closed without merging**